Focus/Iso.swift

```diff
@@ -6,12 +6,21 @@
 //  Copyright (c) 2015 TypeLift. All rights reserved.
 //
 
-/// Captures an isomorphism between S and A.
+/// Captures an isomorphism between S, A and B, T.
 ///
-/// - parameter S: The source of the Iso heading right
-/// - parameter T: The target of the Iso heading left
-/// - parameter A: The source of the Iso heading right
-/// - parameter B: The target of the Iso heading left
+/// In practice, an `Iso` is used with two structures that can be converted between each other 
+/// without information loss.  For example, the Isomorphism between `Optional<T>` and 
+/// `ImplicitlyUnwrappedOptional<T>` is expressed as
+///
+///     Iso<Optional<T>, Optional<T>, ImplicitlyUnwrappedOptional<U>, ImplicitlyUnwrappedOptional<U>>
```

Comment:
This is now wrong in a *different* way.  Transpose the middle `T`-`U` pair; otherwise that type is uninhabitable (modulo bottom).